mars/oscar/backends/mars/pool.py

```diff
@@ -597,56 +603,13 @@ def _parse_config(config: Dict, kw: Dict) -> Dict:
         return kw
 
 
-async def _create_sub_pool(
-        actor_config: ActorPoolConfig,
-        process_index: int,
-        started: multiprocessing.Event):
-    try:
-        env = actor_config.get_pool_config(process_index)['env']
-        if env:
-            os.environ.update(env)
-        pool = await SubActorPool.create({
-            'actor_pool_config': actor_config,
-            'process_index': process_index
-        })
-        await pool.start()
-    finally:
-        started.set()
-    await pool.join()
-
-
-def _start_sub_pool(
-        actor_config: ActorPoolConfig,
-        process_index: int,
-        started: multiprocessing.Event):
-    coro = _create_sub_pool(actor_config, process_index, started)
-    asyncio.run(coro)
-
-
-def _start_sub_pool_in_process(
-        actor_config: ActorPoolConfig,
-        process_index: int,
-        start_method: str = None
-):
-    ctx = multiprocessing.get_context(method=start_method)
-    started = ctx.Event()
-    process = ctx.Process(
-        target=_start_sub_pool,
-        args=(actor_config, process_index, started),
-        name=f'MarsActorPool{process_index}'
-    )
-    process.daemon = True
-    process.start()
-    # wait for sub actor pool to finish starting
-    started.wait()
-    return process
+SubProcessHandle = Union[multiprocessing.Process, 'ray.actor.ActorHandle']
 
 
 @_register_message_handler
 class MainActorPool(ActorPoolBase):
-    __slots__ = '_allocated_actors', '_sub_processes', \
-                '_subprocess_start_method', '_auto_recover', \
-                '_monitor_task', '_on_process_down', '_on_process_recover'
+    __slots__ = '_allocated_actors', '_subprocess_start_method', 'sub_actor_pool_manager', \
```

Comment:
I tried to move these basic classes into `mars/oscar/pool.py`, but it will almostly move everythings to `mars/oscar/pool.py`. Should we do it in this PR?

Reply:
I think so, one backend import something from other backend seems not very natural